### ISSUE TITLE
Investigate strange (false?) assembly warnings

### ISSUE THREAD

**Issue by petemoore** [spectrum4/spectrum4#4]
See [warnings in CI](https://github.com/spectrum4/spectrum4/runs/4776798824?check_suite_focus=true#step:3:992):

```
   552) [0.003s] src/spectrum128k/tests: z80-unknown-elf-as -o test_po_fetch.o test_po_fetch.runner
test_po_fetch.s: Assembler messages:
test_po_fetch.s:44: Warning: 8-bit overflow (-231)
test_po_fetch.s:60: Warning: 8-bit overflow (-163)
test_po_fetch.s:76: Warning: 8-bit overflow (-231)
```

and later on:

```
  546) [0.006s] src/spectrum128k/tests: z80-unknown-elf-as -o test_po_store.o test_po_store.runner
test_po_store.s: Assembler messages:
test_po_store.s:26: Warning: 8-bit overflow (-231)
test_po_store.s:48: Warning: 8-bit overflow (-163)
test_po_store.s:71: Warning: 8-bit overflow (-231)
```

**Comment by petemoore** [comment]
Created https://sourceware.org/bugzilla/show_bug.cgi?id=28791

**Comment by petemoore** [comment]
Should be fixed upstream (according to bug link above), but appears not to be in latest release (2.38). Waiting for a newer release to see if it fixed there. For now leaving this open, until it appears in a new binutils release. I built a new docker image with binutils 2.38 for target z80-unknown-elf locally, and still see the error: 

```bash
 $ cd $(git rev-parse --show-toplevel)/src/spectrum128k/tests; docker run -t --rm -v "$(pwd):/spectrum4" -w /spectrum4 spectrum4:latest /usr/bin/env bash -c 'z80-unknown-elf-as --version && z80-unknown-elf-as test_po_fetch.runner'
GNU assembler (GNU Binutils) 2.38
Copyright (C) 2022 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or later.
This program has absolutely no warranty.
This assembler was configured for a target of `z80-unknown-elf'.
test_po_fetch.s: Assembler messages:
test_po_fetch.s:44: Warning: 8-bit overflow (-231)
test_po_fetch.s:60: Warning: 8-bit overflow (-163)
test_po_fetch.s:76: Warning: 8-bit overflow (-231)
```

**Comment by petemoore** [comment]
It [looks like](https://sourceware.org/pipermail/binutils/2022-July/121567.html) binutils 2.39 is likely to be released in 9 days (Saturday 6 Aug 2022). So we can try again then.

**Comment by petemoore** [comment]
Fixed in 8441d8b3caa232ea0344a41d6a7be517a27bd1c8 with upgrade to binutils 2.39. 🥳
